qiskit/transpiler/passes/optimization/consolidate_blocks.py

```diff
@@ -122,8 +122,14 @@ def run(self, dag):
                 if self.force_consolidate or unitary.num_qubits > 2 or \
                         self.decomposer.num_basis_gates(unitary) != basis_count:
 
-                    new_dag.apply_operation_back(
-                        unitary, sorted(block_qargs, key=lambda x: block_index_map[x]))
+                    if len(block_qargs) <= 2:
+                        new_dag.apply_operation_back(
+                            self.decomposer(unitary).to_gate(),
+                            sorted(block_qargs, key=lambda x: block_index_map[x]))
```

Comment:
Do we ever encounter blocks of width other than 2? Since the blocks should be coming from `Collect2qBlocks`. I agree in the future we will have other widths, but maybe the reason the above works now is because you never encounter blocks of width 1.

Reply:
`ConsolidateBlocks` supports blocks of size larger than two ( https://github.com/Qiskit/qiskit-terra/blob/3c5f86d/test/python/transpiler/test_consolidate_blocks.py#L109 ), but you're right that as we use it now `len(block_qargs)` would always `==2`.